src/msg/simple/Pipe.cc

```diff
@@ -1518,17 +1519,21 @@ void Pipe::stop_and_wait()
 void Pipe::reader()
 {
   pipe_lock.Lock();
-
+  
   if (state == STATE_ACCEPTING) {
     accept();
     assert(pipe_lock.is_locked());
   }
 
+  rcu_register_thread();
+
   // loop.
   while (state != STATE_CLOSED &&
 	 state != STATE_CONNECTING) {
     assert(pipe_lock.is_locked());
 
+    rcu_quiescent_state();
```

Comment:
Same question here, some of the calls in Pipe::reader can block indefinitely, we need to notify the rcu system, right?